### PR TITLE
dynamic_gather::slice_size cannot be of Index type

### PR DESCRIPTION
**Pull request by sdasgup3**

### CODE REVIEW

File: stablehlo/dialect/StablehloOps.td [hunk 1]
```diff
@@ -3449,22 +3449,22 @@ def StableHLO_DynamicGatherOp: StableHLO_Op<"dynamic_gather",
...
+    %slice_sizes = stablehlo.constant dense<[1, 2, 2]> : tensor<3xi64>
     %result = "stablehlo.dynamic_gather"(%operand, %start_indices, %slice_sizes) {
       dimension_numbers = #stablehlo.gather<
         offset_dims = [2, 3],
         collapsed_slice_dims = [0],
         start_index_map = [0, 2],
         index_vector_dim = 2>,
       indices_are_sorted = false
-    } : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi32>) -> tensor<2x3x2x2xi32>
+    } : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi64>) -> tensor<2x3x2x2xi32>
```
Comment (ghpvnist) [inline stablehlo/dialect/StablehloOps.td:3460]: i64 in operand and result too.

File: docs/spec.md [hunk 2]
```diff
@@ -2706,7 +2706,7 @@ op, with the `slice_sizes` specified dynamically as a value.
     start_index_map = [1, 0],
     index_vector_dim = 2>,
   indices_are_sorted = false
-} : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi32>) -> tensor<2x3x2x2xi32>
+} : (tensor<3x4x2xi32>, tensor<2x3x2xi64>, tensor<3xi64>) -> tensor<2x3x2x2xi32>
```
Comment (ghpvnist) [inline docs/spec.md:2709]: i64 in operand and result too.

File: stablehlo/dialect/Base.td [hunk 3]
```diff
@@ -252,6 +252,9 @@ def HLO_PredIntFpOrQuantizedTensor : RankedTensorOf<[HLO_Pred, HLO_Int, HLO_Floa
 // Static representation of a shape vector as a tensor.
 def HLO_StaticDimensionTensor : RankedTensorOf<[HLO_DimensionValue], [HasStaticShapePred, HasAnyRankOfPred<[1]>], "statically shaped 1-dimensional tensor">;
 
+// Static representation of a 1D tensor of int.
+def HLO_StaticIDIntTensor : RankedTensorOf<[HLO_Int], [HasStaticShapePred, HasAnyRankOfPred<[1]>], "statically shaped 1-dimensional tensor">;
```
Comment (ghpvnist) [inline stablehlo/dialect/Base.td:256]: Typo: `HLO_Static1DIntTensor`

**Review by ghpvnist — APPROVED** [review]
LGTM with minor comments

File: stablehlo/dialect/StablehloOps.td (hunk 1 above)
Comment (sdasgup3) [reply stablehlo/dialect/StablehloOps.td]: done.

File: docs/spec.md (hunk 2 above)
Comment (sdasgup3) [reply docs/spec.md]: done

File: stablehlo/dialect/Base.td (hunk 3 above)
Comment (sdasgup3) [reply stablehlo/dialect/Base.td]: done.

File: stablehlo/dialect/Base.td [hunk 4]
```diff
@@ -252,6 +252,9 @@ def HLO_PredIntFpOrQuantizedTensor : RankedTensorOf<[HLO_Pred, HLO_Int, HLO_Floa
 // Static representation of a shape vector as a tensor.
 def HLO_StaticDimensionTensor : RankedTensorOf<[HLO_DimensionValue], [HasStaticShapePred, HasAnyRankOfPred<[1]>], "statically shaped 1-dimensional tensor">;
 
+// Static representation of a 1D tensor of int.
+def HLO_Static1DIntTensor : RankedTensorOf<[HLO_Int], [HasStaticShapePred, HasAnyRankOfPred<[1]>], "statically shaped 1-dimensional tensor">;
```
Comment (mlevesquedion) [inline stablehlo/dialect/Base.td:256]: 1-dimensional *integer* tensor

**Review by mlevesquedion — APPROVED** [review]
Nit: the additional changes in this PR besides defining and using HLO_Static1DIntTensor don't seem related to that change? Consider doing those changes in a separate PR.

**Pull request merged**